scripts/travis_doxygen.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -e
+
+script_folder=`dirname $0`
+pip install --user unidiff
+
+doxygen --version &> /dev/null
+if [[ $? -ne 0 ]]
+then
+  echo "doxygen not installed"
+  echo "Ensure doxygen is installed, then run again"
+  exit 1
+fi
+
+cd src && doxygen &> ../doxygen.log && cd ..
+
+if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+  $script_folder/run_diff.sh 'cat doxygen.log' HEAD~1 # Check for errors introduced in last commit
```

Comment:
The reason is that I wanted to unify cpplint and doxygen checks to avoid code duplication. However, cpplint is currently run on each file separately, whereas doxygen runs on the whole code base. I could run cpplint on the whole code base and pipe the results through the filter (which would probably work as is, except that it has to process more output than necessary).